packages/core/test/app/ovm/examples/simple-state-channel.spec.ts

```diff
@@ -0,0 +1,729 @@
+        aAddress
+      )
+
+      checkSignedMessage(
+        nextMessage,
+        aAddress,
+        new BigNumber(2),
+        parsedMessage.message.channelId,
+        [aAddress],
+        parsedMessage.message.data['addressBalance']
+      )
+    })
+
+    it('handles new message from B on created channel', async () => {
+      const signedMessage: SignedMessage = await createChannel()
+      await acknowledgeMessage(signedMessage, b)
+
+      const addressBalance: AddressBalance = {}
+      addressBalance[bAddress.toString()] = new BigNumber(5)
+      addressBalance[aAddress.toString()] = new BigNumber(15)
+
+      const nextMessage: SignedMessage = await b.createNewMessage(
+        addressBalance,
+        aAddress
+      )
+
+      const parsedMessage: ParsedMessage = parseStateChannelSignedMessage(
+        nextMessage,
+        bAddress
+      )
+
+      checkSignedMessage(
+        nextMessage,
+        bAddress,
+        new BigNumber(2),
+        parsedMessage.message.channelId,
+        [bAddress],
+        parsedMessage.message.data['addressBalance']
+      )
+    })
+
+    it('acknowledges new message from A on created channel', async () => {
+      const signedMessage: SignedMessage = await createChannel()
+      await acknowledgeMessage(signedMessage, b)
+
+      const addressBalance: AddressBalance = {}
+      addressBalance[aAddress.toString()] = new BigNumber(5)
+      addressBalance[bAddress.toString()] = new BigNumber(15)
+
+      const nextMessage: SignedMessage = await a.createNewMessage(
+        addressBalance,
+        bAddress
+      )
+
+      const parsedMessage: ParsedMessage = parseStateChannelSignedMessage(
+        nextMessage,
+        bAddress
+      )
+
+      checkSignedMessage(
+        nextMessage,
+        aAddress,
+        new BigNumber(2),
+        parsedMessage.message.channelId,
+        [aAddress],
+        parsedMessage.message.data['addressBalance']
+      )
+
+      await acknowledgeMessage(nextMessage, b, new BigNumber(2))
+    })
+
+    it('acknowledges new message from B on created channel', async () => {
+      const signedMessage: SignedMessage = await createChannel()
+      await acknowledgeMessage(signedMessage, b)
+
+      const addressBalance: AddressBalance = {}
+      addressBalance[bAddress.toString()] = new BigNumber(5)
+      addressBalance[aAddress.toString()] = new BigNumber(15)
+
+      const nextMessage: SignedMessage = await b.createNewMessage(
+        addressBalance,
+        aAddress
+      )
+
+      const parsedMessage: ParsedMessage = parseStateChannelSignedMessage(
+        nextMessage,
+        bAddress
+      )
+
+      checkSignedMessage(
+        nextMessage,
+        bAddress,
+        new BigNumber(2),
+        parsedMessage.message.channelId,
+        [bAddress],
+        parsedMessage.message.data['addressBalance']
+      )
+
+      await acknowledgeMessage(nextMessage, a, new BigNumber(2))
+    })
+  })
+
+  describe('channel exit', () => {
```

Comment:
This section is the super duper awesome meat of the OVM work here--epic!!!

I think generally, it might be useful to in the future to explicitly/manually construct the exit claims which are being checked behind the scenes here.  Since these decisions are being outsourced to the state channel client, it's a little tricky to parse what's going on between these two files--so it might've been helpful to first write a manual exit decision in the test, so that a reader can follow what's going on "closer to the metal."

But, since this is an example, it's not critical, and not worth redoing that work now--just something to keep in mind for the future. 😃

Reply:
Good call. This definitely makes sense to isolate the test as well. I'll update.